client/state/initial-state.js

```diff
@@ -24,7 +25,8 @@ export const MAX_AGE = 7 * DAY_IN_HOURS * HOUR_IN_MS;
 function getInitialServerState() {
 	// Bootstrapped state from a server-render
 	if ( typeof window === 'object' && window.initialReduxState ) {
-		return window.initialReduxState;
+		const serverState = reducer( window.initialReduxState, { type: SERVER_DESERIALIZE } );
+		return pick( serverState, Object.keys( window.initialReduxState ) );
```

Comment:
Why do you need to `pick` here? Related, if there's a need to trim reduced state to reflect that of the global object, is there not a concern about nested structures?